python/python/tests/test_pubsub.py

```diff
@@ -0,0 +1,1800 @@
+            # So to avoid flakiness, we make sure to unsubscribe from the channels
+            await listening_client_basic.custom_command(
+                ["UNSUBSCRIBE", *list(basic_channels_and_messages.keys())]
+            )
+            await listening_client_pattern.custom_command(["PUNSUBSCRIBE", PATTERN])
+            await listening_client_sharded.custom_command(
+                ["SUNSUBSCRIBE", *list(sharded_channels_and_messages.keys())]
+            )
+
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize(
+        "method", [MethodTesting.Async, MethodTesting.Sync, MethodTesting.Callback]
+    )
+    async def test_pubsub_combined_different_channels_same_name(
+        self, request, cluster_mode: bool, method: MethodTesting
+    ):
+        """
+        Tests combined Pub/Sub with different channel modes using the same channel name.
+        One publishing clients, 3 listening clients, one for each mode.
+
+        This test verifies that separate clients can correctly handle subscriptions for exact, pattern, and sharded channels with the same name.
+        It covers the following scenarios:
+        - Subscribing to an exact channel and verifying message reception.
+        - Subscribing to a pattern channel and verifying message reception.
+        - Subscribing to a sharded channel and verifying message reception.
+        - Ensuring that messages are correctly published and received using different retrieval methods (async, sync, callback).
+        - Verifying that no messages are left unread.
+        - Properly unsubscribing from all channels to avoid interference with other tests.
+        """
+        CHANNEL_NAME = "same-channel-name" + get_random_string(3)
+        MESSAGE_BASIC = get_random_string(10)
+        MESSAGE_PATTERN = get_random_string(7)
+        MESSAGE_SHARDED = get_random_string(5)
+
+        callback, context = None, None
+        callback_messages_basic: List[
+            Union[RedisClient.PubSubMsg, RedisClusterClient.PubSubMsg]
+        ] = []
+        callback_messages_pattern: List[
+            Union[RedisClient.PubSubMsg, RedisClusterClient.PubSubMsg]
+        ] = []
+        callback_messages_sharded: List[
+            Union[RedisClient.PubSubMsg, RedisClusterClient.PubSubMsg]
+        ] = []
+
+        if method == MethodTesting.Callback:
+            callback = new_message
+            context = callback_messages_basic
+
+        # Setup pub/sub for exact channel
+        pub_sub_basic = create_pubsub_subscription(
+            cluster_mode,
+            {PubSubChannelModes.Exact: {CHANNEL_NAME}},
+            callback=callback,
+            context=context,
+        )
+
+        publishing_client, listening_client_basic = await setup_clients(
+            request,
+            cluster_mode,
+            pub_sub_basic,
+        )
+
+        # (Redis version > 7)
+        if await check_if_server_version_lt(publishing_client, "7.0.0"):
```

Comment:
inst it that only the sharded mode should be skipped ?

Reply:
I have one tetst with sharded mode and one without, so we can skip all of the test